sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -492,9 +495,11 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
 
     // If this table is cached as an InMemoryRelation, drop the original
     // cached version and make the new version cached lazily.
+
+    // Uncache the logicalPlan. Note this is a no-op for the table itself if it's not cached, but
+    // will invalidate all caches referencing this table.
+    sparkSession.sharedState.cacheManager.uncacheQuery(table, cascade = true)
```

Comment:
We don't need `blocking = true`?

Reply:
because the default value is already true